mne_bids/report.py

```diff
@@ -419,14 +418,13 @@ def _summarize_channels_tsv(bids_root, scans_fpaths, verbose=True):
 
             # convert to BIDS Path
             params = get_entities_from_fname(bids_basename)
-            bids_basename = BIDSPath(**params)
+            bids_basename = BIDSPath(root=bids_root, **params)
 
             # XXX: improve to allow emptyroom
             if bids_basename.subject == 'emptyroom':
                 continue
 
-            channels_fname = _find_matching_sidecar(bids_fname=bids_basename,
-                                                    bids_root=bids_root,
+            channels_fname = _find_matching_sidecar(bids_path=bids_basename,
```

Comment:
this is awkward ... so `bids_basename` is now an instance of `BIDSPath`? To avoid confusion and be explicit that it's a path object and not str, I would probably do `bids_basename_path`

Reply:
or do we always mean a path object from now on? And it's an `str` only when we do `bids_path.basename` ?

Reply:
Yeah I think the next PR, which I can do today is cosmetic, where we rename everything from `bids_basename` to `bids_path`, and if you want the basename str, then you do `bids_path.basename`.

What do you think?

Reply:
okay, and just to clarify `bids_path` is always a full path for a particular file? or can it also represent a "search path" with unknown entities?

Reply:
It's always represented by root and the basename, which then builds up the full path, but not necessarily for a particular file. For example:

```Python
bids_path = BIDSPath(subject='01', session='02', run='01')
print(bids_path)
>>> /sub-01/ses-02/sub-01_ses-02_run-01

bids_path.update(root='/bids_root')
>>> /bids_root/sub-01/ses-02/sub-01_ses-02_run-01

bids_path.update(suffix='meg')
>>> /sub-01/ses-02/meg/sub-01_ses-02_run-01_meg

bids_path.update(extension='.fif')
>>> /sub-01/ses-02/meg/sub-01_ses-02_run-01_meg.fif
```

The search path will use `_find_matching_sidecar` which relies on the entities stored in BIDSPath and a suffix/extension that overrides what's in BIDSPath.

Reply:
> Yeah I think the next PR, which I can do today is cosmetic, where we rename everything from `bids_basename` to `bids_path`, and if you want the basename str, then you do `bids_path.basename`.

+10

Reply:
There should be some more "sensible" error checks within ` BIDSPath._check()`, but perhaps we add them in as they come up?

For example if:

```Python
bids_path.update(suffix='meg')
>>> /sub-01/ses-02/meg/sub-01_ses-02_run-01_meg

# this should produce an error because it's not allowed
bids_path.update(modality='eeg')
>>> /sub-01/ses-02/eeg/sub-01_ses-02_run-01_meg
```

Reply:
is there a public function that takes an incomplete `bids_path` and returns a list of filenames matching that?

Reply:
No, but... we should! Can you make an issue describing that flow, and we can hash it out there?

Reply:
shouldn't [this](https://mne.tools/mne-bids/dev/generated/mne_bids.path.get_matched_basenames.html#mne_bids.path.get_matched_basenames) take a `bids_path` ?

Reply:
That was my intention originally, but @agramfort suggested we take all entities separately to be closer to how a CLI would work. But good thing you're raising this again, maybe we should reconsider this decision? I'd be +0.75 to allow BIDSPath in `mne_bids.path.get_matched_basenames`